libs/agno/agno/agent/agent.py

```diff
@@ -1826,31 +1852,58 @@ def create_run_response(
         content_type: Optional[str] = None,
         created_at: Optional[int] = None,
         citations: Optional[Citations] = None,
+        run_response: Optional[RunResponse] = None,
     ) -> RunResponse:
-        self.run_response = cast(RunResponse, self.run_response)
         thinking_combined = (thinking or "") + (redacted_thinking or "")
+
+        tools = None
+        response_audio = None
+        audio = None
+        images = None
+        videos = None
+        model = None
+        messages = None
+        extra_data = None
+
+        if run_response:
+            model = run_response.model
+            messages = run_response.messages
+            extra_data = run_response.extra_data
+            content = run_response.content
+            content_type = run_response.content_type
+            audio = run_response.audio
+            images = run_response.images
+            videos = run_response.videos
+            response_audio = run_response.response_audio
+            citations = run_response.citations
+            tools = run_response.tools
+            reasoning_content = run_response.reasoning_content
+
         rr = RunResponse(
             run_id=self.run_id,
             session_id=session_id,
             agent_id=self.agent_id,
             content=content,
             thinking=thinking_combined if thinking_combined else None,
             reasoning_content=reasoning_content,
-            tools=self.run_response.tools,
```

Comment:
Hmm this is potentially a breaking change. Since you don't actually pass run_response on all of the calls to `create_run_response`

Reply:
good catch, passed run_response in all the places where required. Referred a bit from Teams